bigbluebutton-client/src/org/bigbluebutton/modules/chat/views/ChatToolbar.mxml

```diff
@@ -78,6 +90,19 @@
 				globalDispatcher.dispatchEvent(copyEvent);
 			}
 
+			public function sendClearEvent():void{
+				var clearPupUpWarning:IFlexDisplayObject = PopUpManager.createPopUp(FlexGlobals.topLevelApplication as DisplayObject, ClearPublicChatPopUpWarning, true);
+
+				clearPupUpWarning.x = this.localToGlobal(new Point()).x - clearPupUpWarning.width + clearBtn.width + 7;
+				clearPupUpWarning.y = this.localToGlobal(new Point()).y + clearPupUpWarning.height/2 + clearBtn.height + 5;
+			}
+
+			public function onChangeMyRole(e:ChangeMyRole):void{
+				clearBtn.visible = UsersUtil.amIModerator() && publicChat;
+				clearBtn.enabled = UsersUtil.amIModerator() && publicChat;
+				clearBtn.includeInLayout = UsersUtil.amIModerator() && publicChat;
```

Comment:
You can change it to something like:

`clearBtn.visible = clearBtn.includeInLayout = UsersUtil.amIModerator() && publicChat;`